crates/project/src/direnv.rs

```diff
@@ -0,0 +1,71 @@
+use std::{path::Path, process::ExitStatus};
+
+use collections::HashMap;
+
+pub enum DirenvLoadError {
+    Io(std::io::Error),
+    NonZeroResult { status: ExitStatus, stderr: Vec<u8> },
+    Serde(serde_json::Error),
+}
+
+#[derive(Debug, Clone, Default)]
+pub struct DirenvWarning {
+    warning: Option<String>,
+}
+
+impl From<String> for DirenvWarning {
+    fn from(value: String) -> Self {
+        Self {
+            warning: Some(value),
+        }
+    }
+}
+
+impl DirenvWarning {
+    pub fn take(&mut self) -> Option<String> {
+        self.warning.take()
+    }
+}
+
+impl Drop for DirenvWarning {
+    fn drop(&mut self) {
+        if let Some(ref warning) = self.warning {
+            log::warn!("{warning}");
+        }
+    }
+}
```

Comment:
Not a fan of this, this seems overly complicated.

Reply:
Yeah, I've removed this